src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/JitHelper.cs

```diff
@@ -62,16 +62,31 @@ public static void GetEntryPoint(TypeSystemContext context, ReadyToRunHelper id,
                     break;
 
                 case ReadyToRunHelper.WriteBarrier:
-                    mangledName = context.Target.Architecture == TargetArchitecture.ARM64 ? "RhpAssignRefArm64" : "RhpAssignRef";
+                    mangledName = context.Target.Architecture switch
```

Comment:
This method looks quite strange to me.
Is standalone mangled name required for a given architecture?
Why does enum members contain xarch-specific register names?

Reply:
I would definitely drop the suffix in any new code and just name it `RhpAssignRef` like everywhere else. Not sure what was the historical reason for the ARM64 deviation.

Reply:
The write barrier JIT helper uses non-standard calling convention. It is the reason behind the name deviation. We need write barrier helper with standard calling convention that is used in other places and that uses the simple name.

Does Loogarch use non-standard calling convention for write barrier JIT helper as well?

We may be able to simplify this a bit by using the same name for the write barrier JIT helper with custom calling convention on all architectures that have it.

> Why does enum members contain xarch-specific register names?

x86 32-bit has multiple write barrier JIT helpers with custom calling convention for historic reasons.

Reply:
Yes, LoongArch64 use non-standard calling convention for write barrier JIT helper as well. I’ve deleted the useless 
`RhpByRefAssignRefLoongArch64` and `RhpCheckedAssignRefLoongArch64`, but I may not be sure how to simplify 
`RhpAssignRef`.